draft-pauly-oblivious-doh.md

```diff
@@ -132,46 +135,53 @@ via the Oblivious Proxy.
-the Oblivious Target's DoH server is running. See {{request-example}} for an
-example request.
+    RELAY target.example.com:443 HTTP/1.1
+    Host: target.example.com:443
 
-Oblivious DoH messages have no cache value since both requests and responses are
-encrypted using ephemeral key material. Clients SHOULD prefer using HTTP methods and
-headers that will prevent unhelpful cache storage of these exchanges (i.e., preferring POST
-instead of GET).
+When using HTTP/2 {{!H2=RFC7540}} or later, RELAY requests use HTTP pseudo-headers
+with the following requirements:
 
-Clients MUST set the HTTP Content-Type header to "application/oblivious-dns-message"
-to indicate that this request is an Oblivious DoH query intended for proxying. Clients also SHOULD
-set this same value for the HTTP Accept header.
+*  The ":method" pseudo-header field is set to "RELAY".
+*  The ":scheme" pseudo-header field MUST be set to "https".
+*  The ":path" pseudo-header field is set to the path to which the RELAY message
+   body should be sent.
+*  The ":authority" pseudo-header field contains the host and port to
+  connect to (equivalent to the authority-form of the request-target
+  of RELAY requests (see {{!RFC7230}}, Section 5.3)).
 
-Upon receiving a request that contains a "application/oblivious-dns-message" Content-Type,
-the DoH server looks for the "targethost" and "targetpath" variables. If the variables are not
-present, then it is the target of the query, and it can decrypt the query ({{encryption}}).
-If the variables are present, then the DoH server is acting as an Oblivious Proxy.
-If it is a proxy, it is expected to send the request on to the Oblivious
-Target using the URI template constructed as "https://targethost/targetpath".
+A RELAY request that does not conform to these restrictions is malformed
+(see {{!H2}}, Section 8.1.2.6).
 
-## HTTP Request Example {#request-example}
+## RELAY Request {#oblivious-request}
 
-The following example shows how a client requests that an Oblivious Proxy, "dnsproxy.example.net",
-forwards an encrypted message to "dnstarget.example.net". The URI template for the Oblivious
+The following example shows how a client requests that an Oblivious Proxy, "proxy.example.net",
+forwards an encrypted message to "target.example.net". The URI template for the Oblivious
 Proxy is "https://dnsproxy.example.net/dns-query{?targethost,targetpath}". The URI template for
 the Oblivious Target is "https://dnstarget.example.net/dns-query".
 
 ~~~
-:method = POST
+:method = RELAY
 :scheme = https
-:authority = dnsproxy.example.net
-:path = /dns-query?targethost=dnstarget.example.net&targetpath=/dns-query
-accept = application/oblivious-dns-message
+:authority = proxy.example.net
+:path = /dns-query
+accept = application/oblivious-http-message
 cache-control = no-cache, no-store
-content-type = application/oblivious-dns-message
+content-type = application/oblivious-http-message
```

Comment:
This raises the question to me of if other content types would be appropriate for this RELAY method, or if should be using a content type at all here if there is an HTTP method solution. CONNECT doesn't use content type, etc.

Is RELAY only for oblivious stuff? If so, I think it could stand alone.

Reply:
I bet purists will argue that it need not be used for oblivious stuff. Our use case just happens to lump the two together. Are you suggesting that we not use RELAY at all and just use different content types? That seems odd to me, since content type (as I understand it) dictates the format of the request/response and not the semantics for how the request/response is handled.

Reply:
Yeah, this is the question I have. Partly, I'm wondering if this is indeed fundamentally different semantics for how to handle a request/response, vs just a POST that uses a proxy and has an encrypted message format.

Reply:
My take is that it's a different semantics, but I'm not an HTTP expert.

Reply:
Let's chat with @mcmanus too on this

Reply:
Calling @mnot!